test/phase/providers/BUILD.bazel

```diff
@@ -0,0 +1,15 @@
+load(":phase_providers_expose.bzl", "phase_expose_provider_singleton", "rule_that_needs_custom_provider", "scala_library_that_exposes_custom_provider")
+
+scala_library_that_exposes_custom_provider(
+    name = "scala_library_that_exposes_custom_provider",
+)
+
+rule_that_needs_custom_provider(
```

Comment:
Are we just checking we can build this target?

Reply:
Yes. The build failed before because the rule requires the custom provider to be passed and so bazel failed the build. You can take the first commit and see the failure.